### PR TITLE
Add utilities for doing chain operations and misc. other stuff

### PR DESCRIPTION
**Pull request by lovesh**
Signed-off-by: lovesh <lovesh.bond@gmail.com>

### CODE REVIEW

File: src/utils/chain_ops.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+// Convert tokens to micros
+export function docksToMicroDocks(amount) {
+  // Token has `DECIMAL` decimal places
+  const micros = amount * (10 ** DECIMAL);
+  checkValidMicroAmount(micros);
+  return micros;
+}
+
+// Transfer tokens and return txn hash. Converts to mirco tokens first
+export async function transferDocks(api, senderKeypair, recipAddr, amount) {
```
Comment (SamHellawell) [inline src/utils/chain_ops.js:128]: should be `transferDock` not `transferDocks` ? same with below for `MicroDocks`. We would say, for example: "Can you transfer me 1000 DOCK" not "DOCKs"

File: example/chain_ops.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import assert from 'assert';
+import { DockAPI } from '../src/api';
+import {
+  getLastBlock, getBlockNo, blockNumberToHash, getBlock, getBalance, getAllExtrinsicsFromBlock, getTransfersFromBlock, getAllEventsFromBlock, getLastFinalizeBlock, generateAccount, transferMicroDocks, transferDocks,
+} from '../src/utils/chain_ops';
```
Comment (SamHellawell) [inline example/chain_ops.js:5]: can we name this file chain-ops.js to keep in line with naming conventions

File: example/chain_ops.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
+import assert from 'assert';
+import { DockAPI } from '../src/api';
+import {
+  getLastBlock, getBlockNo, blockNumberToHash, getBlock, getBalance, getAllExtrinsicsFromBlock, getTransfersFromBlock, getAllEventsFromBlock, getLastFinalizeBlock, generateAccount, transferMicroDocks, transferDocks,
```
Comment (SamHellawell) [inline example/chain_ops.js:4]: please break down this long line into chunks so thats its more easily readable:
```
getLastBlock, getBlockNo, blockNumberToHash,
getBlock, getBalance, getAllExtrinsicsFromBlock,
...
```

**Review by SamHellawell — CHANGES_REQUESTED** [review]

**Review by SamHellawell — APPROVED** [review]

File: src/utils/chain-ops.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+// The methods below intentionally take an `ApiPromise` object to decouple from DockAPI.
+
+// Generate an account. If `secretUri` is not passed, a random one is generated
+export async function generateAccount({ secretUri, type = 'sr25519', network = 'test' }) {
+  if (['ed25519', 'sr25519', 'ecdsa'].indexOf(type) === -1) {
+    throw new Error(`${type} is not a valid type`);
+  }
+  if (network !== 'test' && network !== 'main') {
+    throw new Error('Network must be "test" or "main" or "dev"');
```
Comment (bddap) [inline src/utils/chain-ops.js:22]: This error still gets thrown when `network === "dev"`

**Review by bddap — APPROVED** [review]
It's a little awkward to be accept a "format" parameter when it's possible for the user to just format the return value. I'm guessing I understand the purpose though, looks like a user ergonomics thing.

The functions in chain-ops serve nicely as documentation outlining what you can do and how.

Thanks for fixing spelling in formatting for that CD stuff.

File: src/utils/chain-ops.js (hunk 4 above)
Comment (lovesh) [reply src/utils/chain-ops.js:22]: Oh, will fix.

**Pull request merged**